persistence/ravendb/clusterconfiguration.md

```diff
@@ -0,0 +1,35 @@
+
+Using cluster-wide transactions correctly is complex. Cluster-wide transactions make use of compare-exchange values in order to guarantee consistent writes across the majority of nodes in the cluster. Each document requires a corresponding compare-exchange value to accomplish that. There’s no mechanism to keep documents and related compare-exchange values in sync, it’s up to the user to manage that. The main challenge is dealing with consistent compare-exchange values when using them to write data.
+
+To provide some context on what that means practically speaking, to enable the usage of cluster wide transactions it's required to:
+- Modify all existing documents to store a synchronization identifier. This can be part of the document, or can be stored inside the document’s metadata.
+- Create a corresponding compare exchange value for every document. The value of the compare exchange value needs to be set to the synchronization identifier’s value.
+Once that’s done, in order to update a document it's needed to:
+- Load the document (and its meta data if needed)
+- Load the corresponding compare exchange value
+- Compare the document’s synchronization identifier with the one stored in the compare exchange value that corresponds to the document. If they are not the same, it means that the document has already been changed and both documents need to be reloaded.
+- Update the document as well as the synchronization identifier
+- Update the compare exchange value’s document value with the new synchronization identifier
+
+Given the complexity involved in implementing cluster wide transactions properly across the business data and the persistence data that’s stored by NServiceBus, it was decided to stop supporting clusters in RavenDB for now. Why? The RavenDB persistence can use cluster wide transactions to correctly and safely store NServiceBus data, but the persistence can’t make sure any business data modified in an NServiceBus handler is safe. Each handler would need to be reassessed using the steps listed above to ensure correct usage of cluster-wide transactions.
+
+Not using cluster-wide transactions in the correct form for the business data would result in possible data loss that is impossible to detect. Given that it may surface as possible message loss, it was decided to withdraw support for clusters. Unless cluster wide transactions were already used in the correct form, introducing them comes with a very severe impact on the system.
```

Comment:
```suggestion
Not using cluster-wide transactions in the correct form for the business data could result in data loss that is impossible to detect. Due to the potential for data loss, support for RavenDB clusters was withdrawn. Unless cluster-wide transactions were already used in the correct form, introducing them comes with a very severe impact on the system.
```

Reply:
Additional note: Nowhere in the document does it state that just enabling cluster-wide transactions removes all of the existing optimistic concurrency checks since CWT does not support normal optimistic concurrency checks, therefore making existing code less safe.